src/main/kotlin/fi/metatavu/oss/api/impl/pages/answers/PageAnswerController.kt

```diff
@@ -331,4 +344,44 @@ class PageAnswerController {
         return "$deviceId-$pageId-$deviceAnswerId"
     }
 
+    /**
+     * Assigns a failsafe answer value to the answer in case it doesn't have one e.g. the answer is malformed
+     *
+     * @param pageQuestion question to get failsafe option for
+     * @param answer answer submitted from the device with malformed data
+     * @return answer with failsafe value
+     */
+    private suspend fun getFailsafeQuestionOption(pageQuestion: PageQuestionEntity, answer: DevicePageSurveyAnswer): DevicePageSurveyAnswer {
+        val dummyAnswer = questionOptionRepository.findFailsafeQuestionOption(pageQuestion.id) ?: createFailsafeQuestionOption(pageQuestion)
+
+        if (pageQuestion.type == FREETEXT) {
```

Comment:
Shouldn't this be done before findFailsafeQuestionOption?

Reply:
Resolved